### PR TITLE
Add component spec documentation

### LINKED ISSUE #64: Create 'Component specification' documentation

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Fixes #64 

Click the `Details` button next to the documentation build step to see the generated docs.

### CODE REVIEW

File: docs/component_spec.md [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+
+### Args
+
+The `args` section describes which arguments the component takes.
+
+```yaml
+args:
+  custom_argument:
+    description: A custom argument
+    type: str
```
Comment (GeorgesLorre) [inline docs/component_spec.md:133]: Should we explicitly mention that these are python types ?

File: docs/component_spec.md [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+A Fondant manifest describes the contents of a dataset and is used as a reference to the dataset
+to be passed between Fondant components.
+
+Creating the manifest is handled by Fondant, as each component will automatically generate the
+manifest that represents its output dataset. While you should not have to create the manifest
+yourself, it is still useful to understand how it works.
+
+## Contents
+
+A manifest consists of the following sections
```
Comment (GeorgesLorre) [inline docs/component_spec.md:20]:
```suggestion
A component spec(ification) consists of the following sections
```

**Review by GeorgesLorre — COMMENTED** [review]
Nice ! The hosted documentation looks amazing. Left some small comments.

File: docs/component_spec.md [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+- Only those fields defined in the `produces` section of the component specification are written 
+  to storage
+
+Each field defines the expected data type, which should match the 
+[types defined by Fondant](https://github.com/ml6team/fondant/blob/main/fondant/schema.py#L13), 
+which mostly match the [Arrow data types](https://arrow.apache.org/docs/python/api/datatypes.html).
+
+#### AdditionalSubsets & additionalFields
+
+The schema also defines the `additionalSubsets` and `additionalFields` keywords, which can be
```
Comment (GeorgesLorre) [inline docs/component_spec.md:110]: nice and clear, especially with the examples

File: docs/component_spec.md [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+used to run it.
+
+```yaml
+name: Example component
+description: This is an example component
+image: example_component:latest
+...
+```
+
+### Consumes & produces
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply docs/component_spec.md:49]: Yes, that's the goal once the components are merged.

File: docs/component_spec.md [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+The `consumes` and `produces` sections describe which data the component consumes and produces. 
+The specification below for instance defines a component that creates an embedding from an 
+image-caption combination.
+
+```yaml
+...
+consumes:
+  images:
+    fields:
+      data:
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply docs/component_spec.md:60]: Very slightly diversified :sweat: 
Agree that it's good to show other possibilities, but also don't want to add noise.

**Pull request merged**